src/elfpy/types.py

```diff
@@ -257,16 +263,18 @@ class RandomSimulationVariables:
 
     # dataclasses can have many attributes
     # pylint: disable=too-many-instance-attributes
-    target_liquidity: float = field(metadata="total size of the market pool (bonds + shares)")
-    target_pool_apr: float = field(metadata="desired fixed apr for as a decimal")
-    fee_percent: float = field(metadata="percent to charge for LPer fees")
-    vault_apr: list = field(metadata="yield bearing source APR")
-    init_vault_age: float = field(metadata="fraction of a year since the vault was opened")
-    init_share_price: float = field(default=None, metadata="initial market share price for the vault asset")
+    target_liquidity: float = field(metadata=to_description("total size of the market pool (bonds + shares)"))
+    target_pool_apr: float = field(metadata=to_description("desired fixed apr for as a decimal"))
+    fee_percent: float = field(metadata=to_description("percent to charge for LPer fees"))
+    vault_apr: list = field(metadata=to_description("yield bearing source APR"))
+    init_vault_age: float = field(metadata=to_description("fraction of a year since the vault was opened"))
+    init_share_price: float = field(
+        default=0, metadata=to_description("initial market share price for the vault asset")
+    )
 
     def __post_init__(self):
         """init_share_price is a function of other random variables"""
-        if self.init_share_price is None:
+        if self.init_share_price == 0:
```

Comment:
Why did you not like the `None` default? After this change, what would a user do if they actually want it to be 0?

Reply:
None can be included in typing using the `Optional` type. https://docs.python.org/3/library/typing.html#typing.Optional

Can you use that and allow the None default?

I would rather not initialize it to a value out of principle, because it is opaque as to whether that value means "not initialized" or whether it is a reasonable initialization.

Reply:
The issue with this is that we then need to check that it isn’t None everywhere that it is used (this is what by the vanilla approach not being typesafe). I don’t think the principle is worth it with this in mind, but let me know if you disagree.